modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -247,7 +247,7 @@ func (h *handler) Trigger(key string, bd *fleet.BundleDeployment) (*fleet.Bundle
 		logrus.Debugf("Adding OnChange for bundledeployment's '%s' resource list", key)
 		return bd, h.trigger.OnChange(key, resources.DefaultNamespace, func() {
 			// enqueue bundledeployment if any resource changes
-			h.bdController.Enqueue(bd.Namespace, bd.Name)
+			h.bdController.EnqueueAfter(bd.Namespace, bd.Name, 0)
```

Comment:
why don't we want to use the rate limit here? what if the `Bundle` is in an error state and this is triggered too many times in a small period of time? 
I think it is safer to use `Enqueue` here, and use the backoff from wrangler

Reply:
It is my understanding this fires every time a _resource_ that was deployed by bundledeployment change.

The reasoning was: if a resource has changed, why should we wait to react to the change? This is not about the bundle changing or error conditions, AFAIU.

Reply:
It waits just if there was an error. The first time it shouldn't wait, the first time it retries it should wait 2^1 * time, the second 2^2 * time, third 2^3 * time and so on.

Reply:
My understanding is that the underlying work queue rate limits depending on its key (the bundle namespace+name in this case) and it has no notion of what a handler `error` is. So it is true all handler `error`s result in Enqueuing, but not all Enqueuings originate from `error`s.

As a consequence, I thought, we should generally use Enqueue in error-like conditions - cases to be treated as if the handler exited with an `error`. OTOH, the reasoning went, we should not use `Enqueue`, but rather `EnqueueAfter` or something else, in case we are triggering something for a condition which is not an `error` - because we do not want it treated like an `error`.

If a bunch of non-error situations happen, I imagined, we do not want to artificially slow down the work queue, but rather work through them as fast as possible. Only if we have errors we want to slow down, because they might be temporary and we do not want to busy wait.

Does the above make any sense to you?

Reply:
it makes sense, thanks for the explanation!